Assets/Scripts/Difficulty/Upgraders/LeafUpgraders/JackolanternSpawnerUpgraders/SpawnTimeMaximumUpgrader.cs

```diff
@@ -0,0 +1,23 @@
+using LimboOfCeres.Scripts.Difficulty.Upgraders.CompositeCore;
+using LimboOfCeres.Scripts.Shared.ScriptableObjectsDefinitions;
+using LimboOfCeres.Scripts.Shared;
+using UnityEngine;
+
+
+namespace LimboOfCeres.Scripts.Difficulty.Upgraders.LeafUpgraders.JackolanternSpawnerUpgraders
+{
+    public class SpawnTimeMaximumUpgrader : LeafUpgrader
+    {
+        [SerializeField]
+        private JackolanternSpawnerScriptable spawnerData;
+        private const float LevelUpFactor = 0.9f;
+
+        protected override void OnUpgradeHook()
+        {
+            spawnerData.SpawnTimeMaximum.LimitedValue *= LevelUpFactor;
+        }
+
+        public override bool IsAtLimit => Mathf.Approximately(
+            Constants.Spawners.Jackolanterns.SpawnTimeMaximum.Minimum, spawnerData.SpawnTimeMaximum.LimitedValue);
```

Comment:
Could use the minimum / maximum properties in the scriptable object. This happens on most other upgraders.